### PR TITLE
[WIP] #36gjt0 http proxy-server and image 

### PR DESCRIPTION
**Pull request by andrewelkin**
k8s auth service [https://app.clickup.com/1250339/v/li/14840533]

### CODE REVIEW

File: proxy-server/test/auth_server.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+from urllib.parse import parse_qs
+import logging
+import os
+from sys import argv
+import jwt
+from datetime import datetime, timezone
+
+debug_key = "debugKeyForTestOnlydNeverUseInProduction123456789012345678901234567890"
+
+""" @Alex Foran: This is a part for backend.  It should redirect to url from get_redirect_url()
```
Comment (andrewelkin) [inline proxy-server/test/auth_server.py]: @forana here is the code for backend

File: proxy-server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM golang:1.8-alpine
+
+RUN apk --no-cache add git
+
+WORKDIR /go/src/proxy-main
+ADD ./go/proxy-main.go ./
+
+RUN go get -v
+RUN go install -v .
```
Comment (forana) [inline proxy-server/Dockerfile]: Someday, I'd like to convert this to a gomod and freeze off the dependency versions.

File: proxy-server/go/proxy-main.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+const keyLength = 512 / 8
+
+const (
+	errGeneric          = iota
+	errNoCookie         = iota
+	errBadSignature     = iota
+	errExpired          = iota
+	errTokenInvalid     = iota
+	errSignatureInvalid = iota
```
Comment (forana) [inline proxy-server/go/proxy-main.go]: you can omit the `iota` after all but the first of these. https://github.com/golang/go/wiki/Iota

File: proxy-server/go/proxy-main.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+		Value:   tokenString,
+		Expires: expirationTime,
+	})
+}
+
+// The default fallback handler.
+func handleRequestFallBack(res http.ResponseWriter, req *http.Request) {
+	res.WriteHeader(http.StatusUnauthorized)
+	//		res.WriteHeader(http.StatusBadRequest)
+	fmt.Fprintf(res, "\nWe are sorry, access denied.")
```
Comment (forana) [inline proxy-server/go/proxy-main.go]: This should get sent in the response, not the log.

File: proxy-server/go/proxy-main.go [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+			res.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		if validateAuthToken(claims) {
+			setNewCookie(res, req)
+			log.Printf("OK: Setting cookie")
+			serveReverseProxy(targetURL, res, req)
+			log.Printf("OK: Proxying to url: %s", targetURL)
+		}
```
Comment (forana) [inline proxy-server/go/proxy-main.go]: should have a `return` after this

File: proxy-server/go/proxy-main.go (hunk 4 above)
Comment (andrewelkin) [reply proxy-server/go/proxy-main.go]: This _is_ a response ! :)  (Writer)

Comment (forana) [reply proxy-server/go/proxy-main.go]: Oh! Yes it is :)

File: proxy-server/go/proxy-main.go (hunk 5 above)
Comment (andrewelkin) [reply proxy-server/go/proxy-main.go]: >should have a return after this
naked return? yes, but  there is no return value for this function..

**Comment by hhuuggoo** [comment]
Btw. This repo is public.  I'm not concerned about this work going here right now but a separate repo is probably appropriate.  Just make sure we don't check in any secrets 😉

**Pull request merged**